gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -343,11 +369,34 @@ gpii.oauth2.authServer.buildAuthorizedServicesPayload = function (authorizationS
             serviceName: client.clientName
         };
     });
-    return {
-        username: user.username,
+
+    promiseTogo.resolve({
+        username: userData.username,
         authorizedServices: authorizedServices,
         unauthorizedServices: unauthorizedServices
-    };
+    });
+
+    return promiseTogo;
+};
+
+gpii.oauth2.authServer.buildAuthorizedServicesPayload = function (authorizationService, user) {
+    // TODO: Update the user interface to support multiple tokens per
+    // user rather than using a single default
+    var gpiiToken = user.defaultGpiiToken;
+
+    var authorizedClientsPromise = authorizationService.getAuthorizedClientsForGpiiToken(gpiiToken);
+    var unauthorizedClientsPromise = authorizationService.getUnauthorizedClientsForGpiiToken(gpiiToken);
+
+    // TODO: Update the usage of fluid.promise.sequence() once https://issues.fluidproject.org/browse/FLUID-5938 is resolved.
+    var sources = [authorizedClientsPromise, unauthorizedClientsPromise];
+    var promisesSequence = fluid.promise.sequence(sources);
+
+    var authorizedServicesPromise = fluid.promise();
+    promisesSequence.then(function (responses) {
```

Comment:
Add a note to convert this to fluid.promise.map once we implement https://issues.fluidproject.org/browse/FLUID-5968